io4edge_client/binaryiotypeb/client.py

```diff
@@ -0,0 +1,156 @@
+        """
+        fs_response = Pb.ConfigurationDescribeResponse()
+        self._fb_client.describe(Pb.ConfigurationDescribe(), fs_response)
+        return fs_response
+
+    def set_output(self, channel: int, state: bool):
+        """
+        Set the state of a single output.
+        @param channel: channel number
+        @param state: state to set. a "true" state turns on the outputs switch, a "false" state turns it off.
+        @raises RuntimeError: if the command fails
+        @raises TimeoutError: if the command times out
+        """
+        fs_cmd = Pb.FunctionControlSet()
+        fs_cmd.single.channel = channel
+        fs_cmd.single.state = state
+        self._fb_client.function_control_set(fs_cmd, Pb.FunctionControlSetResponse())
+
+    def set_all_outputs(self, states: int, mask: int):
+        """
+        Set the state of all or a group of output channels.
+        @param states: binary coded map of outputs. 0 means switch off, 1 means switch on, LSB is Channel0
+        @param mask: binary coded map of outputs to be set. 0 means do not change, 1 means change, LSB is Channel0
+        @raises RuntimeError: if the command fails
+        @raises TimeoutError: if the command times out
+        """
+        fs_cmd = Pb.FunctionControlSet()
+        fs_cmd.all.values = states
+        fs_cmd.all.mask = mask
+        self._fb_client.function_control_set(fs_cmd, Pb.FunctionControlSetResponse())
+
+    def exit_error_state(self):
+        """
+        Try to recover the binary output controller from error state.
+        The binary output controller enters error state when there is an overurrent condition for a long time.
+        In the error state, no outputs can be set; inputs can still be read.
+        This call tells the binary output controller to try again.
+        This call does however not wait if the recovery was successful or not.
+        @raises RuntimeError: if the command fails
+        @raises TimeoutError: if the command times out
+        """
+        fs_cmd = Pb.FunctionControlSet()
+        fs_cmd.exit_error.CopyFrom(Pb.SetExitError())
+        self._fb_client.function_control_set(fs_cmd, Pb.FunctionControlSetResponse())
+
+    def input(self, channel: int) -> bool:
+        """
+        Get the state of a single channel, regardless whether its configured as input or output)
+        State "true" is returned if the input level is above switching threshold, "false" otherwise.
+        @param channel: channel number
+        @return: state of the input
+        @raises RuntimeError: if the command fails
+        @raises TimeoutError: if the command times out
+        """
+        fs_cmd = Pb.FunctionControlGet()
+        fs_cmd.single.channel = channel
+        fs_response = Pb.FunctionControlGetResponse()
+        self._fb_client.function_control_get(fs_cmd, fs_response)
+        return fs_response.single.state
+
+    def all_inputs(self, mask: int) -> int:
```

Comment:
I would remove it. We can add it later when we have a binaryIoTypeB hardware that supports inputs